### ISSUE TITLE
Using par() and effect size color on plots

### ISSUE THREAD

**Issue by lpaulopires** [ACCLAB/dabestr#53]
Hi guys!

I'm currently changing some of my analysis for a publication using estimation statistics and estimation plots.
Firstly, it is fantastic!

I could use some help on two little issues, hope they are not naive enough (can't say I'm an R-expert).

1) I'm trying to use par(mfrow()) to plot multiple plots side-by-side but I've failed so far. Can't figure out why. Could you guys help me with this one?

2) The default color for the bootstrap sampling distribution in Gardner–Altman plots is grey and I can't find a way to change this. However, in Fig. 1e (Ho et al. 2019) it is red. Was it changed manually in another software? I read that color setting is an issue for v0.3.0, but if there is an alternative available, it would be of great help.

Thank you!

**Comment by josesho** [comment]
Sorry for the delay in replying, @lpaulopires. I've been out of the office for the past 2 weeks.

>1. I'm trying to use par(mfrow()) to plot multiple plots side-by-side but I've failed so far. Can't figure out why. Could you guys help me with this one?

We actually use [`cowplot`](https://github.com/wilkelab/cowplot) to stitch two `ggplot2` objects: the rawdata plot, and the effect size plot. It turns out, you can do the same with `cowplot::plot_grid()`!!

```r
library(dabestr)

estplot1 <- dabest(iris, Species, Petal.Width,
                   idx = c("setosa", "versicolor")) %>% plot()

estplot2 <- dabest(iris, Species, Petal.Width,
                   idx = c("versicolor", "virginica")) %>% plot()

combi.plot <- cowplot::plot_grid(estplot1, estplot2)
```

![combiplot](https://user-images.githubusercontent.com/12838917/63489281-40303700-c4e4-11e9-8802-1129f2ce1eab.png)

_The caveat is, we can't set a common palette across different plots yet. This is ongoing work, again for v0.3.0 hopefully._

> 2. The default color for the bootstrap sampling distribution in Gardner–Altman plots is grey and I can't find a way to change this. However, in Fig. 1e (Ho et al. 2019) it is red. Was it changed manually in another software? I read that color setting is an issue for v0.3.0, but if there is an alternative available, it would be of great help.

As you mentioned, this is currently in the works. Unfortunately there is no code-based alternative for now :( 
FYI Figure 1e in [Ho et al. 2019](https://rdcu.be/bHhJ4) was generated with [DABEST-Python](https://github.com/ACCLAB/DABEST-python), which was developed almost a year before this R package and thus has extended features. 

Hope this helps!

**Comment by josesho** [comment]
Hi @lpaulopires , 

Thanks for being patient! This should be resolved by v0.2.5 (PR #77)

Here's a quick recipe for what you want to achieve:

```r
library(dabestr)

# Create a named list to pass as a palette.
my.pal <- setNames(c("red", "black", "green"), 
                           unique(iris$Species))

estplot1 <- dabest(iris, Species, Petal.Width,
                   idx = c("setosa", "versicolor")) %>% 
            plot(color.column = Species, 
                 palette = head(my.pal, 2), # only get the first two colors.
                 rawplot.ylim = c(0, 2.6))

estplot2 <- dabest(iris, Species, Petal.Width,
                   idx = c("versicolor", "virginica")) %>% 
            plot(color.column = Species, 
                 palette = tail(my.pal, 2), # only get the last two colors.
                 rawplot.ylim = c(0, 2.6))

combi.plot <- cowplot::plot_grid(estplot1, estplot2)

combi.plot # show the plot
``` 

![image](https://user-images.githubusercontent.com/12838917/79843995-a1026d00-83ed-11ea-9076-75def13a8f83.png)
